autosklearn/estimators.py

```diff
@@ -112,22 +113,37 @@ def __init__(
             Incompatible with include. Include and exclude are incompatible
             if used together on the same component
 
-        resampling_strategy : string or object, optional ('holdout')
-            how to to handle overfitting, might need 'resampling_strategy_arguments'
-
-            * 'holdout': 67:33 (train:test) split
-            * 'holdout-iterative-fit':  67:33 (train:test) split, calls iterative
-              fit where possible
-            * 'cv': crossvalidation, requires 'folds'
-            * 'cv-iterative-fit': crossvalidation, calls iterative fit where possible
-            * 'partial-cv': crossvalidation with intensification, requires
-              'folds'
-            * BaseCrossValidator object: any BaseCrossValidator class found
-                                        in scikit-learn model_selection module
-            * _RepeatedSplits object: any _RepeatedSplits class found
-                                      in scikit-learn model_selection module
-            * BaseShuffleSplit object: any BaseShuffleSplit class found
-                                      in scikit-learn model_selection module
+        resampling_strategy : Union[str, BaseCrossValidator, _RepeatedSplits, BaseShuffleSplit] = "holdout"
+            How to to handle overfitting, might need to use ``resampling_strategy_arguments``
+            if using ``"cv"`` based method or a Splitter object.
+
+            If using a Splitter object that relies on the dataset retaining it's current
+            size and order, you will need to look at the ``dataset_compression`` argument
+            and ensure that ``"subsample"`` is not included in the applied compression
+            ``"methods"`` or disable it entirely with ``False``.
+
+            **Options**
+
+            *   ``"holdout"``:
+                    67:33 (train:test) split
+            *   ``"holdout-iterative-fit"``:
+                    67:33 (train:test) split, iterative fit where possible
+            *   ``"cv"``:
+                    crossvalidation,
+                    requires ``"folds"`` in ``resampling_strategy_arguments``
+            *   ``"cv-iterative-fit"``:
+                    crossvalidation,
+                    calls iterative fit where possible,
+                    requires ``"folds"`` in ``resampling_strategy_arguments``
+            *   'partial-cv':
+                    crossvalidation with intensification,
+                    requires ``"folds"`` in ``resampling_strategy_arguments``
+            *   ``BaseCrossValidator`` object:
+                    any BaseCrossValidator class found in scikit-learn model_selection module
+            *   ``_RepeatedSplits`` object:
+                    any _RepeatedSplits class found in scikit-learn model_selection module
+            *   ``BaseShuffleSplit`` object:
+                    any BaseShuffleSplit class found in scikit-learn model_selection module
```

Comment:
Seeing this, it could also be a custom subclass of them, right?

Reply:
True, I'll add a little more documentation to say it can be a subclass of any of those.